semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -223,7 +223,8 @@ trait TextDocumentOps { self: SemanticdbOps =>
                   }
                 }
               } else {
-                occurrences(mtree.pos) = symbol
+                val selectionFromStructuralType = gsym.owner.isRefinementDummy
+                if (!selectionFromStructuralType) occurrences(mtree.pos) = symbol
```

Comment:
It took me a while to understand the motivation for this change but I think I get it now. Structural types can be produced on the fly so the structure `{ def foo: Int }` does not have a global unique symbols. I agree with the strategy to edge on the conservative side and not emit occurrences here, poking around IntelliJ it seems to struggle with structural types as well

![2018-06-06 13 24 53](https://user-images.githubusercontent.com/1408093/41040531-f9ebbb82-699c-11e8-9e22-12db67f20516.gif)

In the long term, we need to come up with a solution for this case since it's highly relevant for tools to know which occurrences are selections on structural types. One crazy idea might be to introduce `SymbolOccurrence.Role.STRUCTURAL` and make `SymbolOccurrence.symbol` be empty. The structural type could be embedded in the `SymbolOccurrence.tpe` of the prefix. Anyways, we can leave this discussion to an issue to unblock this PR. This commit LGTM 👍

Reply:
Can we open a ticket with the description of this commit + add a link in the code to the issue? It's not clear from the source code why we skip on `selectionFromStructuralType`

Reply:
https://github.com/scalameta/scalameta/issues/1591